packages/magento-product/components/AddProductsToCart/useAddProductsToCartAction.ts

```diff
@@ -33,6 +37,31 @@ export function useAddProductsToCartAction(
     loading,
   } = props
 
+  const [showSuccess, setShowSuccess] = useState<boolean>(false)
+
+  const userErrors = toUserErrors(data)
+
+  const submitSuccesful =
+    !formState.isSubmitting &&
+    formState.isSubmitSuccessful &&
+    !error?.message &&
+    !userErrors.length &&
+    !redirect
+
+  useEffect(() => {
+    if (submitSuccesful && submittedVariables?.cartItems.find((item) => item.sku === sku)) {
+      setShowSuccess(true)
+    }
+  }, [sku, submitSuccesful, submittedVariables?.cartItems])
```

Comment:
We might be able to do this without useEffect: https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state